client/lib/catch-js-errors/index.js

```diff
@@ -1,81 +1,113 @@
-				localStorage.setItem( 'log-errors', 'external' );
-				//Prep the stage up for Google Stackdriver or other service experiment
-			} else {
-				localStorage.setItem( 'log-errors', 'false' );
+			}
+
+			if ( errorLogger === 'rest-api' ) {
+				// set up handler to POST errors
+				TraceKit.report.subscribe( errorReport => {
+					const error = {
+						message: errorReport.message,
+						url: document.location.href,
+						trace: errorReport.stack,
+					};
+
+					this.diagnose();
+					this.sendToApi( Object.assign( error, this.diagnosticData ) );
+				} );
+			} else if ( errorLogger === 'analytics' ) {
+				TraceKit.report.subscribe( function gaApiLogger( errorReport ) {
+					if ( typeof window.ga === 'function' ) {
+						window.ga(
+							'send',
+							'event',
+							'JS Error',
+							'URL: ' + document.location.href,
+							errorReport.message + ' ' + JSON.stringify( errorReport.stack ),
+							0,
+							{ nonInteraction: true }
+						);
+					}
+				} );
 			}
 		}
+	}
 
-		if ( errorLogger === 'true' ) {
-			//Fix up any outstanding old settings
-			localStorage.setItem( 'log-errors', 'rest-api' );
-			errorLogger = 'rest-api';
+	saveDiagnosticReducer( data ) {
+		this.diagnosticReducers.push( data );
+	}
+
+	saveDiagnosticData( data ) {
+		if ( typeof data.extra === 'object' ) {
+			this.saveExtraData( data );
+		} else {
+			Object.assign( this.diagnosticData, data );
 		}
+	}
 
-		if ( errorLogger === 'rest-api' ) {
-			// set up handler to POST errors
-			TraceKit.report.subscribe( function restApiLogger( errorReport ) {
-				const error = {
-					message: errorReport.message,
-					url: document.location.href,
-					trace: errorReport.stack,
-				};
+	saveExtraData( data ) {
+		Object.assign( this.diagnosticData.extra, data );
+	}
 
-				if ( window.navigator && window.navigator.userAgent ) {
-					error.browser = window.navigator.userAgent;
-				}
+	diagnose() {
+		this.diagnosticReducers.forEach( diagnosticReducer => {
+			try {
+				this.saveDiagnosticData( diagnosticReducer() );
+			} catch ( e ) {
+				console.warn( 'diagnostic', this.diagnosticData );
+			}
+		} );
 
-				sendToApi( error );
-			} );
-		} else if ( errorLogger === 'analytics' ) {
-			TraceKit.report.subscribe( function gaApiLogger( errorReport ) {
-				if ( typeof window.ga === 'function' ) {
-					window.ga(
-						'send',
-						'event',
-						'JS Error',
-						'URL: ' + document.location.href,
-						errorReport.message + ' ' + JSON.stringify( errorReport.stack ),
-						0,
-						{ nonInteraction: true }
-					);
-				}
-			} );
-		}
+		return this.diagnosticData;
+	}
+
+	sendToApi( error ) {
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'POST', 'https://public-api.wordpress.com/rest/v1.1/js-error?http_envelope=1', true );
+		xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
+		let params = 'client_id=39911&client_secret=cOaYKdrkgXz8xY7aysv4fU6wL6sK5J8a6ojReEIAPwggsznj4Cb6mW0nffTxtYT8&error=';
```

Comment:
Oh sorry, realized this was still in the standalone lib. Ignore the comment above.